Licensing v2 (#8519)
harness/developer-hub: docs/continuous-delivery/get-started/service-licensing-for-cd.md

- The ‘Instance Fetch’ script is properly configured and Harness has steady visibility to all SIs for the service.

```TEXT
Harness CD consumes 1 Service License for every 20 SIs of the custom service.
```

- The ‘Instance Fetch’ script is not properly set up or functioning as expected - Harness has no visibility to SIs for the service.

```TEXT
Harness CD consumes 1 Service License for each active custom service.
```

### Pipelines with no Service

Harness allows custom deployments, where no service is associated with the deployment. This can happen when a pipeline execution only runs infrastructure provisioning steps, only performs shell script executions, or runs a custom stage with the environment configured, but no service. In all these scenarios, lack of service config means Harness loses the default license tracking. In these scenarios:

```TEXT
Harness CD consumes 1 Service License for every 100 pipeline executions of such custom stages.
```

## Subscription Page Walkthrough

*Update: 11/06/2024* 

> Harness has launched an improved service tracking change to track services deployed and managed by Harness. With the launch of improved tracking, we have also launched a new subscription page allowing users to visualize what is being tracked by Harness.

Reviewer: Harness has introduced an enhanced service tracking feature to monitor services deployed and managed by Harness. Alongside this update, we've added a new subscription page where users can easily view all tracked services.

| **GitOps Service**           | Not measurable by customer, not calculated in product | 1 GitOps Service = 20 SI (1 Service)            | New service tracking introduced for GitOps Services.                                                                                 |
| **Pipeline Execution (Deployments w/o Service)** | Not measurable by customer, not calculated in product | 2000 Executions = 1 Service                     | New service tracking introduced for Deployments without service. Specifically, per Stage Execution. If a Pipeline has 5 Stages with no service being deployed, we bill 5 Executions. |
| **Serverless Services**      | 1 Function = 1 Service                          | 1 Function = 0.2 Service For SI based - No Charge | Adjusted valuation to 0.2 Service per function.                                                                                       |

## Customer FAQs for the improved Licensing Tracking

> ### Q1: Will Licensing V2 increase my costs?

Reviewer: ```suggestion
### Will Licensing V2 increase my costs?
```

> **A:** Potentially, yes. Your license consumption may increase since we are now tracking more deployable items and offering additional value. This could result in a higher renewal cost. However, we are committed to working with you to adjust your contract to match your usage and needs. We are not using these increases to charge you more; we are just trying to track license utilization across all customers better.

Reviewer: ```suggestion
Potentially, yes. Your license consumption may increase since we are now tracking more deployable items and offering additional value. This could result in a higher renewal cost. However, we are committed to working with you to adjust your contract to match your usage and needs. We are not using these increases to charge you more; we are just trying to track license utilization across all customers better.
```

> ### Q2: Why has tracking for GitOps Services been introduced in Licensing V2?

Reviewer: ```suggestion
### Why has tracking for GitOps Services been introduced in Licensing V2?
```

> **A:** GitOps has become a core feature of Harness Continuous Deployment (CD) and is widely adopted by our customers. Including GitOps Services in the billing model ensures that billing accurately reflects the comprehensive set of features you utilize, promoting fairness and transparency.

Reviewer: ```suggestion
GitOps has become a core feature of Harness Continuous Deployment (CD) and is widely adopted by our customers. Including GitOps Services in the billing model ensures that billing accurately reflects the comprehensive set of features you utilize, promoting fairness and transparency.
```

> ### Q3: How does the change in Serverless Services valuation benefit me?

Reviewer: ```suggestion
### How does the change in Serverless Services valuation benefit me?
```

> **A:** By adjusting the valuation to 0.20 of a Service per serverless function, customers deploying serverless architectures will experience reduced costs associated with these functions. This change makes it more cost-effective for you to leverage serverless deployments.

Reviewer: ```suggestion
By adjusting the valuation to 0.20 of a Service per serverless function, customers deploying serverless architectures will experience reduced costs associated with these functions. This change makes it more cost-effective for you to leverage serverless deployments.
```

> ### Q4: Will my overall billing increase with Licensing V2?
> 
> **A:** It depends on your usage patterns. You may see little to no change if you primarily use CD Services with minimal Deployments without service and do not utilize GitOps or Deployment Templates. However, if you heavily use the newly billed components, your billing may increase accordingly. We recommend reviewing your usage and considering optimization strategies to manage costs effectively.

Reviewer: ```suggestion
```

Reviewer: IMO, redundant of Q1

### Q5: How can I calculate my expected billing under Licensing V2?

**A:** You can estimate your total billed services by summing up your usage based on the new licensing rules:

> - **Service Instances:** Count the Service Instances for CD and GitOps Services.

Reviewer: This doesn't seem very clear, and since this whole page is about billing, its redundant IMO.

> ### Q5: How can I calculate my expected billing under Licensing V2?
> 
> **A:** You can estimate your total billed services by summing up your usage based on the new licensing rules:
> 
> - **Service Instances:** Count the Service Instances for CD and GitOps Services.

Reviewer: ```suggestion
```

> ### Q6: Does Harness provide tools to help manage and optimize licensing costs?
> 
> **A:** Yes, Harness offers monitoring and reporting tools within the platform to help you track usage across all licensing components. You can generate detailed reports based on the Service Licensing Page to monitor your consumption and identify areas for optimization.

Reviewer: ```suggestion
### Does Harness provide tools to help manage and optimize licensing costs?

Yes, Harness offers monitoring and reporting tools within the platform to help you track usage across all licensing components. You can generate detailed reports based on the Service Licensing Page to monitor your consumption and identify areas for optimization.
```

> ### Q7: What is happening with the User License I initially renewed or purchased?
> 
> **A:** The User License is now considered a legacy license type that we no longer support for Harness CD. While you can renew based on the User License, the licensing page will compute usage in terms of services, not users, and you will not receive an accurate estimate of user counts. Originally, Harness CD tracked deployed services rather than users. The User License model was created to offer more favorable pricing terms for some customers.

Reviewer: ```suggestion
### What will happen with the User License I initially renewed or purchased?

The User License is now considered a legacy license type that we no longer support for Harness CD. While you can renew based on the User License, the licensing page will compute usage in terms of services, not users, and you will not receive an accurate estimate of user counts. Originally, Harness CD tracked deployed services rather than users. The User License model was created to offer more favorable pricing terms for some customers.
```

> ### Q8: What is happening with the Service or Service Instance Licenses I initially renewed or purchased?
> 
> **A:** Harness will migrate existing customers with Service or Service Instance Licenses to the V2 version of these license types. After discussing with you, the licensing page will compute usage according to the V2 licensing model. Please note that you cannot renew under the legacy V1 Service or Service Instance Licenses.
> 

Reviewer: ```suggestion
### What will happen with the Service or Service Instance Licenses I initially renewed or purchased?

Harness will migrate existing customers with Service or Service Instance Licenses to the V2 version of these license types. After discussing with you, the licensing page will compute usage according to the V2 licensing model. Please note that you cannot renew under the legacy V1 Service or Service Instance Licenses.

```

> ### Q9: I'm concerned about unexpected cost increases. How can I ensure my costs remain manageable?
> 
> **A:** We understand concerns about cost management. This change is not intended to upcharge you. It’s intended to improve service tracking. We will not upcharge based on increased utilization. To help you, we offer tools within the platform that provide detailed visibility into your usage. You can monitor your consumption in real-time and set up alerts for usage thresholds. Additionally, our team is available to work with you to analyze your usage patterns and identify opportunities for optimization to keep your costs predictable.

Reviewer: ```suggestion
### I'm concerned about unexpected cost increases. How can I ensure my costs remain manageable?

We understand concerns about cost management. This change is not intended to upcharge you. It’s intended to improve service tracking. We will not upcharge based on increased utilization. To help you, we offer tools within the platform that provide detailed visibility into your usage. You can monitor your consumption in real-time and set up alerts for usage thresholds. Additionally, our team is available to work with you to analyze your usage patterns and identify opportunities for optimization to keep your costs predictable.
```

> ### Q10: Will there be a grace period to adjust to the new licensing model?
> 
> **A:** We are committed to making this transition as smooth as possible. While your current contract remains unchanged until renewal, we encourage you to review your usage using the new licensing page. This will give you ample time to adjust your workflows and optimize usage before the new licensing terms take effect upon renewal.

Reviewer: ```suggestion
### Will there be a grace period to adjust to the new licensing model?

We are committed to making this transition as smooth as possible. While your current contract remains unchanged until renewal, we encourage you to review your usage using the new licensing page. This will give you ample time to adjust your workflows and optimize usage before the new licensing terms take effect upon renewal.
```

> ### Q11: Can I stay on my current licensing model instead of moving to Licensing V2?
> 
> **A:** Licensing V2 is designed to provide a more accurate and fair billing model based on actual usage. While we recommend transitioning to take advantage of the enhanced features and support, we understand that each customer has unique needs. Please discuss your specific situation with your account representative to explore possible options.

Reviewer: ```suggestion
### Can I stay on my current licensing model instead of moving to Licensing V2?

Licensing V2 is designed to provide a more accurate and fair billing model based on actual usage. While we recommend transitioning to take advantage of the enhanced features and support, we understand that each customer has unique needs. Please discuss your specific situation with your account representative to explore possible options.
```

> ### Q12: How will the licensing changes affect my existing integrations and workflows?
> 
> **A:** The licensing changes are billing-related and should not directly impact your existing integrations and workflows. All your current configurations will continue to function as before. The main difference will be in how usage is tracked and billed. If you have concerns, our support team is ready to assist you in reviewing your setup.

Reviewer: ```suggestion
### How will the licensing changes affect my existing integrations and workflows?

The licensing changes are billing-related and should not directly impact your existing integrations and workflows. All your current configurations will continue to function as before. The main difference will be in how usage is tracked and billed. If you have concerns, our support team is ready to assist you in reviewing your setup.
```

> ### Q13: What support is available to help me transition to Licensing V2?
> 
> **A:** We offer comprehensive support to assist you during this transition:
> 
> - **Dedicated Account Representative:** Your account manager can provide personalized guidance.
> - **Technical Support:** Our technical team can help optimize your configurations for the new licensing model.
> - **Documentation and Resources:** Access detailed guides and best practices within our Help Center.
> - **Webinars and Workshops:** Participate in upcoming sessions focused on navigating Licensing V2.

Reviewer: ```suggestion
### What support is available to help me transition to Licensing V2?

We offer comprehensive support to assist you during this transition:

- **Dedicated Account Representative:** Your account manager can provide personalized guidance.
- **Technical Support:** Our technical team can help optimize your configurations for the new licensing model.
- **Documentation and Resources:** Access detailed guides and best practices within our Help Center.
- **Webinars and Workshops:** Participate in upcoming sessions focused on navigating Licensing V2.
```

> ### Q14: How does Licensing V2 handle seasonal or fluctuating usage patterns?
> 
> **A:** Licensing V2 is designed to accommodate varying usage levels:
> 
> - **Flexible Billing:** The model accounts for actual usage, which can adapt to your seasonal demands.
> - **Scaling Options:** Easily scale up or down based on your needs without long-term commitments for unused capacity.
> - **Usage Monitoring:** Utilize our tools to predict and plan for high or low-activity periods.

Reviewer: ```suggestion
### How does Licensing V2 handle seasonal or fluctuating usage patterns?

Licensing V2 is designed to accommodate varying usage levels:

- **Flexible Billing:** The model accounts for actual usage, which can adapt to your seasonal demands.
- **Scaling Options:** Easily scale up or down based on your needs without long-term commitments for unused capacity.
- **Usage Monitoring:** Utilize our tools to predict and plan for high or low-activity periods.
```

> ### Q15: What if my usage exceeds my licensed amount before renewal?
> 
> **A:** Please contact us if you anticipate exceeding your licensed usage before renewal. We can discuss options such as adjusting your license tier or exploring short-term accommodations to ensure uninterrupted service while we work together on a long-term solution.

Reviewer: ```suggestion
### What if my usage exceeds my licensed amount before renewal?

Please contact us if you anticipate exceeding your licensed usage before renewal. We can discuss options such as adjusting your license tier or exploring short-term accommodations to ensure uninterrupted service while we work together on a long-term solution.
```

> ### Q16: I'm not using certain features, such as GitOps or Serverless Functions. Can my licensing be adjusted accordingly?
> 
> **A:** Absolutely. Licensing V2 is usage-based, so you will only be billed for the components you utilize. If you do not use features like GitOps or Serverless Functions, they will not contribute to your license consumption. This model ensures you pay only for what you use.

Reviewer: ```suggestion
### I'm not using certain features, such as GitOps or Serverless Functions. Can my licensing be adjusted accordingly?

Absolutely. Licensing V2 is usage-based, so you will only be billed for the components you utilize. If you do not use features like GitOps or Serverless Functions, they will not contribute to your license consumption. This model ensures you pay only for what you use.
```

> ### Q17: How does Licensing V2 impact security and compliance features?
> 
> **A:** All security and compliance features remain fully supported and will not be affected by the licensing changes. Your access to these critical features will continue, ensuring your deployments remain secure and compliant with industry standards.

Reviewer: ```suggestion
### How does Licensing V2 impact security and compliance features?

All security and compliance features remain fully supported and will not be affected by the licensing changes. Your access to these critical features will continue, ensuring your deployments remain secure and compliant with industry standards.
```

> ### Q18: Will there be changes to how users are managed and billed in Licensing V2?
> 
> **A:** Under Licensing V2, we have shifted from user-based CD service billing. While we no longer bill based on the number of users, user management functionalities remain unchanged. You can continue to add and manage users within your organization without it affecting your CD licensing costs.

Reviewer: ```suggestion
### Will there be changes to how users are managed and billed in Licensing V2?

Under Licensing V2, we have shifted from user-based CD service billing. While we no longer bill based on the number of users, user management functionalities remain unchanged. You can continue to add and manage users within your organization without it affecting your CD licensing costs.
```

> ### Q19: How can I provide feedback about Licensing V2?
> 
> **A:** We welcome your feedback as it helps us improve our services. You can provide feedback through:
> 
> - **Account Representative:** Share your thoughts directly with your account manager.
> - **Support Tickets:** Submit feedback via our support portal.
> - **Surveys:** Participate in customer satisfaction surveys when prompted.

Reviewer: ```suggestion
### How can I provide feedback about Licensing V2?

We welcome your feedback as it helps us improve our services. You can provide feedback through:

- **Account Representative:** Share your thoughts directly with your account manager.
- **Support Tickets:** Submit feedback via our support portal.
- **Surveys:** Participate in customer satisfaction surveys when prompted.
```

> ### Q20: What if I disagree with the usage data shown on the new licensing page?
> 
> **A:** If you believe there is a discrepancy in your usage data:
> 
> - **Review Details:** Check the detailed reports available to understand the metrics.
> - **Contact Support:** Reach out to our support team for assistance.
> - **Investigation:** We will investigate and resolve any inaccuracies promptly.

Reviewer: ```suggestion
### What if I disagree with the usage data shown on the new licensing page?

If you believe there is a discrepancy in your usage data:

- **Review Details:** Check the detailed reports available to understand the metrics.
- **Contact Support:** Reach out to our support team for assistance.
- **Investigation:** We will investigate and resolve any inaccuracies promptly.
```

> ### Q21: Are there any changes to the terms and conditions in Licensing V2?
> 
> **A:** Licensing V2 may include updates to terms and conditions to reflect the new billing model. We recommend reviewing the updated terms provided during renewal. Your account representative can also walk you through any changes and address your questions.

Reviewer: ```suggestion
### Are there any changes to the terms and conditions in Licensing V2?

Licensing V2 may include updates to terms and conditions to reflect the new billing model. We recommend reviewing the updated terms provided during renewal. Your account representative can also walk you through any changes and address your questions.
```

> ### Q22: How does Licensing V2 affect multi-account or multi-project setups?
> 
> **A:** Licensing V2 provides aggregated usage tracking across all your accounts and projects, offering a unified view of your consumption. This helps in managing licenses more efficiently in complex organizational structures. Billing will reflect the total usage across all entities.

Reviewer: ```suggestion
### How does Licensing V2 affect multi-account or multi-project setups?

Licensing V2 provides aggregated usage tracking across all your accounts and projects, offering a unified view of your consumption. This helps in managing licenses more efficiently in complex organizational structures. Billing will reflect the total usage across all entities.
```

> ### Q23: Can I customize my licensing plan to fit my organization's needs better?
> 
> **A:** We offer flexible licensing options to accommodate various requirements. Please discuss your specific needs with your account representative, and we will tailor a licensing plan that aligns with your operational and budgetary goals.

Reviewer: ```suggestion
### Can I customize my licensing plan to fit my organization's needs better?

We offer flexible licensing options to accommodate various requirements. Please discuss your specific needs with your account representative, and we will tailor a licensing plan that aligns with your operational and budgetary goals.
```

> ### Need Further Assistance?

Reviewer: ```suggestion
## Need Further Assistance?
```
